internal/docstore/drivertest/drivertest.go

```diff
@@ -1382,40 +1381,86 @@ func testUnorderedActions(t *testing.T, coll *ds.Collection, revField string) {
 	}
 }
 
-func testAs(t *testing.T, coll *ds.Collection, st AsTest) {
-	docs := []*HighScore{
-		{game3, "steph", 24, date(4, 25), nil},
-		{game3, "mia", 99, date(4, 26), nil},
+// Verify that BeforeDo is invoked, and its as function behaves as expected.
+func testBeforeDo(t *testing.T, newHarness HarnessMaker) {
```

Comment:
I think this is a more straightforward and comprehensive way to test escape hatches like `BeforeDo` than the way we do it elsewhere, where each driver must implement an `AsTest` interface and write its own tests.

@vangent WDYT?

Reply:
I think this way you save the work of implementing `AsTest` by writing the `XXXTypes` function.

Reply:
It seems fine. You lose the ability to do an end-to-end test of `As` -- for example, `gcsblob` uses `As` to set a field that's not supported by `blob` in the `BeforeWrite`, and then reads it back using the `As` for read. If you don't want to do that, this is more succinct.